[PL-Doc] Delegate Upgrader with Proxy Settings (#10931)
harness/developer-hub: docs/platform/delegates/manage-delegates/configure-delegate-proxy-settings.md

Reviewer: Change line 11, to make it comprehensive and not only limited about delegate as we have added support in upgrader as well.

---

> title: Proxy Configuration Guide.

Reviewer: ```suggestion
title: Proxy configuration guide
```

  -e MANAGER_HOST_AND_PORT=https://<YOUR_MANAGER_HOST_AND_PORT>/delegate:23.09.80505
```

> ### Proxy Configuration for Kubernetes Delegate Upgrader

Reviewer: @ivikramsahu ensure that the headings throughout the doc are consistent.

:::note

>   This feature is available from 1.7.0 and later.

Reviewer: @ivikramsahu please add that the feature would be available from Delegate Upgrader version 1.7.0. Readers might not understand 1.7.0 in isolation. Also, include link to the release note.

Reviewer: This should not come here under a sample config. Please either add this to the top or create a dedicated section for proxy settings for  Upgrader and add this note and other samples under it.

:::

> To configure a proxy for your Kubernetes Delegate Upgrader, ensure that you update the manifest file. Download the `harness-delegate.yaml`, find the `kind: ConfigMap` section, and add your proxy settings (i.e., proxyHost, proxyPort, proxymanager, proxyUser, ProxyPassword) as shown below.

Reviewer: ```suggestion
To configure proxy for your Kubernetes Delegate Upgrader, add the proxy settings to the Delegate upgrader config in the manifest file. Below is an example for the same:
```

Reviewer: The first sentence and the initial bit of the second sentence are redundant. Alsi, **find the `kind: ConfigMap` section** is wrong. There are be multiple config maps in a manifest file. We need to ensure proxy settings are added to the upgrader config.

```yaml
  apiVersion: v1
  kind: ConfigMap
  metadata:
    name: kubernetes-delegate-upgrader-config
    namespace: harness-delegate-ng
  data:
    config.yaml: |
      mode: Delegate
      dryRun: false
      workloadName: kubernetes-delegate
      namespace: harness-delegate-ng
      containerName: delegate
      delegateConfig:
        accountId: px7xd_XXXXXXX_Vjvw

>         managerHost: https://vanityurl.harness.io

Reviewer 2: Do not add a direct example.

Author: Agree, @bharatgoelharness—noted! I think we should include the example, and I’ve masked the information wherever required. This example gives users clearer guidance on how to add proxy settings.

I have now updated it with dummy information in this commit. -> [4493858](https://github.com/harness/developer-hub/pull/10931/commits/449385865ae694021ed64acea83b587a84ec3efc)

      proxyHost: XX.XX.XX.XX
      proxyPort: 3128
      proxyManager: true
      proxyUser: MYUSER
      proxyPassword: ******
```

Once updated, apply the configuration using the command below.

```bash
kubectl apply -f harness-delegate.yaml
```

### Proxy Configuration for Docker Delegate Upgrader

> To run the Docker Delegate Upgrader with proxy settings, use the command below with the required environment variables:

Reviewer: We should not say that use the below command. This is just an example that we are sharing.
